common/src/main/java/com/microsoft/identity/common/internal/cache/MsalOAuth2TokenCache.java

```diff
@@ -552,6 +552,35 @@ public ICacheRecord load(@NonNull final String clientId,
                 null // not applicable
         );
 
+        if (refreshTokens.isEmpty()) {
+            // If we didn't find an RT in the cache, this could be a "TSL-seed" or "dual-client stack"
+            // scenario
+            //
+            // Defining these terms:
+            // TSL-seed: another 1P TSL integrated app has put a token into our cache so we can
+            //     pick it up
+            //
+            // Dual-Client stack: two FoCI-enabled app registrations are sharing a single binary
+            //     and accordingly, can share RTs.
+            //     Examples for this might be TFL/TFW - which uses multiple client ids to enable
+            //     different scenarios depending on enterprise vs. consumer usage
+
+            // Unlike the broker, where we check if an app is FoCI prior to making a network call
+            // with an arbitrary FoCI RT we find in the cache, if we're in standalone mode and find
+            // a FoCI RT in the cache, the current app must also be FoCI (!!!)
+            //
+            // Making the assumption that the current client id can use any FoCI RT we find in the
+            // cache is strictly contingent that app developers NOT mix FoCI/non-FoCI registrations
+            // into same binary. If you do this, you'll get confusing errors that the RT used doesn't
+            // match the client app registration. This assumption means we don't need to implement
+            // "FoCI probing" and/or track FoCI app meta
+            final Credential fallbackFrt = getFamilyRefreshTokenForAccount(account);
+
+            if (null != fallbackFrt) {
+                refreshTokens.add(fallbackFrt);
```

Comment:
Is this a safe copy that we're adding to, with guaranteed mutability, for all implementations of the cache interface?  One would think, but the interface documentation is vague on what we're getting back here other than that it implements List.  Should we take a copy before we mutate it or relax the final modifier on the refreshTokens reference so we can just reassign it since we only want the first element anyway?

Reply:
The `List` is mutable, but as you mention, the interface is vague on this point.

I'll take the suggestion to remove the `final` modifier and will reassign the `List` -- API vagueries should ideally be ironed out as well so I'll spot check classes and make updates to the javadoc where appropriate as part of this PR

There's probably >1 interface floating around here that is ambiguous on the mutability/immutability of the resulting Collection. As you see them, feel free to file an issue (or plural issues) if you see more places like that this that need clarity and those can be addressed either piecemeal or as part of a larger effort

Reply:
Java/mutability clarifed in:
- 7df4abf2dc56e75d737ab1c7f6c94c99fe8e4c81

List reinit + add (even tho we know List is mutable) in:
- a1b4f1b113bf53c57c349ab7bb114dcd7e83a965

Strictly speaking we could take the attitude that the changes in 7df4abf2dc56e75d737ab1c7f6c94c99fe8e4c81 make a1b4f1b113bf53c57c349ab7bb114dcd7e83a965 unnecessary, but I'm fine to incorporate that original suggestion and leave as-is